### PR TITLE
Make it compile with latest Zig version on Linux

### PR DESCRIPTION
**Pull request by locriacyber**
Also, it will now print help if no argument is given.

### CODE REVIEW

File: lib/buzz_os.zig [hunk 1]
```diff
@@ -494,7 +494,7 @@ export fn SocketServerStart(vm: *api.VM) c_int {
 
     server.listen(list.addrs[0]) catch |err| {
         switch (err) {
-            error.NoDevice => vm.bz_pushErrorEnum("lib.errors.FileSystemError", "lib.errors.FileSystemError".len, "NoDevice", "NoDevice".len),
```
Comment (giann) [inline lib/buzz_os.zig:497]: did you run this against zig master? this is why I added this case but they maybe reverted this on later commits?

File: src/obj.zig [hunk 2]
```diff
@@ -295,11 +295,7 @@ pub const ObjFiber = struct {
     };
 };
 
-pub const pcre_struct = switch (builtin.os.tag) {
```
Comment (giann) [inline src/obj.zig:298]: This maybe depends on you're pcre install? it tested this on ubuntu-latest without any issue. Anyway you should not remove the macos, etc. case

**Comment by giann** [comment]
Thanks for the contribution. Try to split the build fix and the help message bit in two commits.

File: lib/buzz_os.zig (hunk 1 above)
Comment (giann) [reply lib/buzz_os.zig:497]: you can see if failed in the github action

Comment (locriacyber) [reply lib/buzz_os.zig:497]: with version 0.10.0-dev.4247+3234e8de3

File: src/obj.zig (hunk 2 above)
Comment (locriacyber) [reply src/obj.zig:298]: I use Arch Linux. I think `pcre_*` should be used with `pcre` type.

```
PCRE_EXP_DECL int  pcre_exec(const pcre *, const pcre_extra *, PCRE_SPTR,
                   int, int, int, int *, int);
PCRE_EXP_DECL int  pcre16_exec(const pcre16 *, const pcre16_extra *,
                   PCRE_SPTR16, int, int, int, int *, int);
PCRE_EXP_DECL int  pcre32_exec(const pcre32 *, const pcre32_extra *,
                   PCRE_SPTR32, int, int, int, int *, int);
```

Comment (giann) [reply src/obj.zig:298]: are you sure you not using pcre2 and not pcre?

Comment (locriacyber) [reply src/obj.zig:298]: It's not pcre2.

**Comment by giann** [comment]
Feel free to reopen if you manage to pass the repo tests

**Pull request closed without merging**